### ISSUE TITLE
pipenv install --skip-lock    produces error message: "An error occurred while installing time!  Will try again"

### ISSUE THREAD

**Issue by cpoptic** [pypa/pipenv#2020]
Describe the issue briefly here.

I have tried uninstalling and reinstalling pipenv
I have tried uninstalling and reinstalling pip

Running Windows 10, 64 bit
I have anaconda and pip installed on C drive (SSD).
My projects directory is on D drive (HD, more space)

I try to create a new virtual environment in my D:\projects directory:

```console
cd d:\projects\myNewProject
pipenv install
```

This produces error:

```console
> Pipfile.lock not found, creating…                                                                                       
Locking [dev-packages] dependencies…
> Locking [packages] dependencies…
> Warning: Your dependencies could not be resolved. You likely have a mismatch in your sub-dependencies.
>   You can use $ pipenv install --skip-lock to bypass this mechanism, then run $ pipenv graph to inspect the situation.
> Could not find a version that matches basemap
> Tried: (no version found at all)
```

I then follow directions and run: `pipenv install --skip-lock`

```console
> Installing dependencies from Pipfile…
> An error occurred while installing time! Will try again.
> An error occurred while installing calendar! Will try again.
> An error occurred while installing codecs! Will try again.
> An error occurred while installing json! Will try again.
> An error occurred while installing sys! Will try again.
> An error occurred while installing gzip! Will try again.
> An error occurred while installing string! Will try again.
> An error occurred while installing glob! Will try again.
> An error occurred while installing re! Will try again.
> An error occurred while installing matplotlib.pyplot! Will try again.
>   ================================ 37/37 - 00:00:07                                                                     An error occurred while installing basemap! Will try again.
> Installing initially–failed dependencies…                                                                               Collecting time --------------------- 0/11 - 00:00:00
>                                                                                                                           Could not find a version that satisfies the requirement time (from -r C:\Users\Chris\AppData\Local\Temp\pipenv-sfeah4r9-requirements\pipenv-l0yoygj0-requirement.txt (line 1)) (from versions: )                                              No matching distribution found for time (from -r C:\Users\Chris\AppData\Local\Temp\pipenv-sfeah4r9-requirements\pipenv-l0yoygj0-requirement.txt (line 1))
> 
>      ================================ 0/11 - 00:00:00
```

I'm not sure why pipenv can't install these dependencies and what specifically I need to do to get them installed.

============================================================

Please run `$ python -m pipenv.help`, and paste the results here.

D:\projects\cs231n.github.io>python -m pipenv.help
<details><summary>$ python -m pipenv.help output</summary>

Pipenv version: `'11.10.0'`

Pipenv location: `'C:\\ProgramData\\Anaconda3\\lib\\site-packages\\pipenv'`

Python location: `'C:\\ProgramData\\Anaconda3\\python.exe'`

Other Python installations in `PATH`:

  - `2.7`: `C:\Python27\python.exe`
  - `3.6`: `C:\ProgramData\Anaconda3\python.exe`

  - `3.6.3`: `C:\ProgramData\Anaconda3\python.exe`
  - `2.7.2`: `C:\Python27\python.exe`
  - `2.7.2`: `C:\Windows\py.exe`

PEP 508 Information:

```
{'implementation_name': 'cpython',
 'implementation_version': '3.6.3',
 'os_name': 'nt',
 'platform_machine': 'AMD64',
 'platform_python_implementation': 'CPython',
 'platform_release': '10',
 'platform_system': 'Windows',
 'platform_version': '10.0.16299',
 'python_full_version': '3.6.3',
 'python_version': '3.6',
 'sys_platform': 'win32'}
```

System environment variables:

  - `ALLUSERSPROFILE`
  - `APPDATA`
  - `ASL.LOG`
  - `COMMONPROGRAMFILES`
  - `COMMONPROGRAMFILES(X86)`
  - `COMMONPROGRAMW6432`
  - `COMPUTERNAME`
  - `COMSPEC`
  - `CONDA_SHLVL`
  - `CONFIGSETROOT`
  - `CUDA_PATH`
  - `CUDA_PATH_V9_0`
  - `GDAL_DATA`
  - `HADOOP_HOME`
  - `HOMEDRIVE`
  - `HOMEPATH`
  - `LOCALAPPDATA`
  - `LOGONSERVER`
  - `MKL_THREADING_LAYER`
  - `MSMPI_BIN`
  - `NUMBER_OF_PROCESSORS`
  - `NVCUDASAMPLES9_0_ROOT`
  - `NVCUDASAMPLES_ROOT`
  - `NVTOOLSEXT_PATH`
  - `ONEDRIVE`
  - `OPENSSL_CONF`
  - `OS`
  - `PATH`
  - `PATHEXT`
  - `POSTGIS_ENABLE_OUTDB_RASTERS`
  - `POSTGIS_GDAL_ENABLED_DRIVERS`
  - `PROCESSOR_ARCHITECTURE`
  - `PROCESSOR_IDENTIFIER`
  - `PROCESSOR_LEVEL`
  - `PROCESSOR_REVISION`
  - `PROGRAMDATA`
  - `PROGRAMFILES`
  - `PROGRAMFILES(X86)`
  - `PROGRAMW6432`
  - `PROMPT`
  - `PSMODULEPATH`
  - `PUBLIC`
  - `PYSPARK_DRIVER_PYTHON`
  - `PYSPARK_DRIVER_PYTHON_OPTS`
  - `PYTHONIOENCODING`
  - `SPARK_HOME`
  - `SYSTEMDRIVE`
  - `SYSTEMROOT`
  - `TEMP`
  - `TMP`
  - `USERDOMAIN`
  - `USERDOMAIN_ROAMINGPROFILE`
  - `USERNAME`
  - `USERPROFILE`
  - `VBOX_MSI_INSTALL_PATH`
  - `VS140COMNTOOLS`
  - `WINDIR`
  - `PYTHONDONTWRITEBYTECODE`
  - `PIP_PYTHON_PATH`
  - `PYTHONUNBUFFERED`

Pipenv–specific environment variables:

Debug–specific environment variables:

  - `PATH`: `C:\ProgramData\Anaconda3\Library\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.0\bin;D:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.0\bin;D:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.0\libnvvp;C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC;D:\utils\cuda;C:\Program Files\Microsoft MPI\Bin\;C:\ProgramData\Oracle\Java\javapath;C:\Program Files (x86)\Intel\iCLS Client\;C:\Program Files\Intel\iCLS Client\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Program Files\dotnet\;C:\Program Files\nodejs\;C:\Program Files\erl9.2\bin;C:\Program Files (x86)\Elixir\bin;C:\Users\Chris\.mix\escripts;C:\ProgramData\Anaconda3;C:\ProgramData\Anaconda3\Scripts;C:\Python27;C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin;C:\Program Files\Calibre2\;C:\Program Files (x86)\Gow\bin;D:\utils\spark-2.2.1-bin-hadoop2.7\bin;C:\Program Files\CMake\bin;C:\Program Files\Git\cmd;C:\Users\Chris\AppData\Local\Microsoft\WindowsApps;C:\Users\Chris\AppData\Roaming\npm;C:\Users\Chris\AppData\Local\atom\bin`

---------------------------

Contents of `Pipfile` ('D:\\projects\\Pipfile'):

```toml
[[source]]
url = "https://pypi.python.org/simple"
verify_ssl = true
name = "pypi"

[packages]
jupyterlab = "*"
watermark = "*"
flask = "*"
numpy = "*"
matplotlib = "*"
tqdm = "*"
"eli5" = "*"
yellowbrick = "*"
seaborn = "*"
datetime = "*"
dask = "*"
pandas = "*"
cython = "*"
pandas-profiling = "*"
forestci = "*"
"bs4" = "*"
nltk = "*"
lxml = "*"
shapely = "*"
gensim = "*"
time = "*"
calendar = "*"
codecs = "*"
json = "*"
sys = "*"
gzip = "*"
string = "*"
glob = "*"
re = "*"
"matplotlib.pyplot" = "*"
tweepy = "*"
networkx = "*"
basemap = "*"
studioml = "*"
tensorflow = "*"
tensorflow-probability = "*"
"176c765" = {path = ".\\dvc"}

[dev-packages]

```

</details>

------------

##### Expected result

Expected a new pipenv virtual environment to be created for my new projeect

##### Actual result

Error message:

> Could not find a version that satisfies the requirement time (from -r C:\Users\Chris\AppData\Local\Temp\pipenv-sfeah4r9-requirements\pipenv-l0yoygj0-requirement.txt (line 1)) (from versions: )
> No matching distribution found for time (from -r C:\Users\Chris\AppData\Local\Temp\pipenv-sfeah4r9-requirements\pipenv-l0yoygj0-requirement.txt (line 1))

##### Steps to replicate

Provide the steps to replicate (which usually at least includes the commands and the Pipfile).

```
> pipenv install --skip-lock
Installing dependencies from Pipfile…
An error occurred while installing time! Will try again.
An error occurred while installing calendar! Will try again.
An error occurred while installing codecs! Will try again.
An error occurred while installing json! Will try again.
An error occurred while installing sys! Will try again.
An error occurred while installing gzip! Will try again.
An error occurred while installing string! Will try again.
An error occurred while installing glob! Will try again.
An error occurred while installing re! Will try again.
An error occurred while installing matplotlib.pyplot! Will try again.
  ================================ 37/37 - 00:00:07
An error occurred while installing basemap! Will try again.
Installing initially–failed dependencies…
Collecting time --------------------- 0/11 - 00:00:00

  Could not find a version that satisfies the requirement time (from -r C:\Users\Chris\AppData\Local\Temp\pipenv-sfeah4r9-requirements\pipenv-l0yoygj0-requirement.txt (line 1)) (from versions: )
No matching distribution found for time (from -r C:\Users\Chris\AppData\Local\Temp\pipenv-sfeah4r9-requirements\pipenv-l0yoygj0-requirement.txt (line 1))

     ================================ 0/11 - 00:00:00
```

**Comment by techalchemy** [comment]
Are you out of disk space on your `C:` drive? It is still putting the virtualenv there.  You can try running `set PIPENV_VENV_IN_PROJECT=1` before running this.

I honestly don't really know anything at all about anaconda distributions of pipenv (does anyone? @uranusjr ? I now @acdha was just working on something with this but I'm not sure if he is able to help at all) but I suspect we are running up against some kind of resolution issue related to the fact that we call pip to do installations... Can you try this with `pipenv install --verbose` and give us the output?

**Comment by cpoptic** [comment]
> Pipfile.lock not found, creating…
> Locking [dev-packages] dependencies…
> Using pip: -i https://pypi.python.org/simple
> 
>                           ROUND 1
> Current constraints:
> 
> Finding the best candidates:
> 
> Finding secondary dependencies:
> ------------------------------------------------------------
> Result of round 1: stable, done
> 
> Locking [packages] dependencies…
> Using pip: -i https://pypi.python.org/simple
> 
>                           ROUND 1
> Current constraints:
>   basemap
>   bs4
>   calendar
>   codecs
>   cython
>   dask
>   datetime
>   eli5
>   flask
>   forestci
>   gensim
>   glob
>   gzip
>   json
>   jupyterlab
>   lxml
>   matplotlib
>   matplotlib.pyplot
>   networkx
>   nltk
>   numpy
>   pandas
>   pandas-profiling
>   re
>   seaborn
>   shapely
>   string
>   studioml
>   sys
>   tensorflow
>   tensorflow-probability
>   time
>   tqdm
>   tweepy
>   watermark
>   yellowbrick
> 
> Finding the best candidates:
> Please check your version specifier and version number. See PEP440 for more information.
> Using pip: -i https://pypi.python.org/simple
> 
>                           ROUND 1
> Current constraints:
>   basemap
>   bs4
>   calendar
>   codecs
>   cython
>   dask
>   datetime
>   eli5
>   flask
>   forestci
>   gensim
>   glob
>   gzip
>   json
>   jupyterlab
>   lxml
>   matplotlib
>   matplotlib.pyplot
>   networkx
>   nltk
>   numpy
>   pandas
>   pandas-profiling
>   re
>   seaborn
>   shapely
>   string
>   studioml
>   sys
>   tensorflow
>   tensorflow-probability
>   time
>   tqdm
>   tweepy
>   watermark
>   yellowbrick
> 
> Finding the best candidates:
> Please check your version specifier and version number. See PEP440 for more information.
> 
> INFO:pip9._vendor.requests.packages.urllib3.connectionpool:Starting new HTTPS connection (1): pypi.org
> Warning: Your dependencies could not be resolved. You likely have a mismatch in your sub-dependencies.
>   You can use $ pipenv install --skip-lock to bypass this mechanism, then run $ pipenv graph to inspect the situation.
> Could not find a version that matches basemap
> Tried: (no version found at all)
> Warning: Your dependencies could not be resolved. You likely have a mismatch in your sub-dependencies.
>   You can use $ pipenv install --skip-lock to bypass this mechanism, then run $ pipenv graph to inspect the situation.
> Could not find a version that matches basemap
> Tried: (no version found at all)

**Comment by techalchemy** [comment]
Can you run `pip install basemap` successfully?

**Comment by cpoptic** [comment]
Also I don't think it has to do with Anaconda; as I've installed pipenv using `pip install pipenv` not using conda.

The C drive is only 100GB so I've installed only programs on there and left the 1TB D drive for the data & projects.  But since the environment variables are set and point to the virtual environment directory on the C drive correctly, it should be fine?

Running `pip install basemap`   yield

> Collecting basemap
>   Could not find a version that satisfies the requirement basemap (from versions: )
> No matching distribution found for basemap

Hmm, looks like under "versions" there's nothing following that colon.  Think that's related to the reason basemap is not installing?

**Comment by uranusjr** [comment]
Basemap is an [externally hosted package](https://reinout.vanrees.org/weblog/2013/02/28/pypi-discussion.html) ([click on the “download” button and you’ll be redirected to SourceForge](https://pypi.org/project/basemap/1.0.7/)). Its [support has been removed](https://www.python.org/dev/peps/pep-0470/) in the new PyPI. [Contact Basemap developers](https://github.com/matplotlib/basemap/issues) to ask them to upload files to PyPI.

Note: I think it is possible to use SourceForge as a custom PyPI, but I’m not interested in experiementing. Basemap devs should figure this out.

**Comment by techalchemy** [comment]
Yes. It's the exact same thing the pipenv error message says.  I asked you to do this because I was already suspecting that you can't actually install anything, due to either a connection problem or an SSL problem or some other issue . This is a bigger problem than pipenv, you aren't able to install packages in your system at all.  Pipenv is never going to work under these circumstances.  There were some issues with pip 10 on windows -- I would recommend pinning 9.0.3 for the moment with `python -m pip install pip==9.0.3` and trying again

**Comment by techalchemy** [comment]
You can use the `PIP_FIND_LINKS` environment variable or whatever, but yeah, definitely not a pipenv issue

**Comment by cpoptic** [comment]
Weird, what do you suspect is the underlying issue then, if not pipenv?

**Comment by techalchemy** [comment]
as @uranusjr mentioned, this is a package that is not on pypi, so you can't simply `pip install` it and you can't `pipenv install` it

**Comment by uranusjr** [comment]
@cpoptic The issue is Basemap. They are using an old way (discontinued 4 years ago!) to point pip to the download link that has been slated for removal *years* ago. The old method is finally removed with the new PyPI launch (pypi.org), making the download link no longer visible to PyPI-based package managers, i.e. pip (and by extension, Pipenv).

There are two solutions I can think of:

* Install the package directly, without going through PyPI. Download the source [here](https://sourceforge.net/projects/matplotlib/files/matplotlib-toolkits/basemap-1.0.7/) and run `pipenv install ..../basecamp-1.0.7.tar.gz`.
* Tell Basecamp to upload the tarball (and maybe wheels) to PyPI, and/or provide a clear alternative.

**Comment by cpoptic** [comment]
@uranusjr Thanks, you're right looks like the last Basemap was released in 2013!

So ultimately, I'm curious what is unique about my setup that's causing this issue, whereas it appears no one else has encountered it?  Surely others have encountered this error involving random package Basemap?

I'm trying to set up everything to be consistent with a reasonably "standard" pipeline that everyone else uses (yes herd mentality can help at times) simply to make troubleshooting easier.  Where am I going wrong, and what can I do to standardize my pipenv setup/configuration?

**Comment by techalchemy** [comment]
Basemap isn’t a random package. It’s a specific package. This is the first time I’ve ever heard of it. It hasn’t been updated in 5 years, and it’s not on pythons main package index. This is not a common problem because mostly people aren’t trying to install this.

**Comment by techalchemy** [comment]
The only thing ‘wrong’ about your setup is that you’re trying to install this particular package
